site/content/blog/2024-07-17-script-scan-rules/index.md

```diff
@@ -0,0 +1,222 @@
+---
+title: "ZAP Scripts are now Full Scan Rules!"
+summary: "ZAP scripts can now do everything that scan rules can."
+images:
+- https://www.zaproxy.org/blog/2024-07-17-script-scan-rules/images/cover.jpg
+type: post
+tags:
+  - blog
+  - scanrules
+  - scripts
+date: "2024-07-17"
+authors:
+  - akshath
+---
+
+ZAP active and passive scripts are now first-class scan rules! 🎉
+
+ZAP scripts may now optionally be exposed as scan rules, allowing you to include them in your scan policies.
+
+Existing scripts are not affected by this change.
+
+## How to Expose a Script as a Scan Rule
+
+Any active or passive script that implements the `getMetadata` function will be exposed as a scan rule to ZAP.
+
+```js
+const ScanRuleMetadata = Java.type("org.zaproxy.addon.commonlib.scanrules.ScanRuleMetadata");
+
+function getMetadata() {
+	return ScanRuleMetadata.fromYaml(`
+id: 12345
+name: Passive Vulnerability Title
+description: Full description
+solution: The solution
+references:
+  - Reference 1
+  - Reference 2
+risk: INFO  # info, low, medium, high
+confidence: LOW  # false_positive, low, medium, high, user_confirmed
+cweId: 0
+wascId: 0
+alertTags:
+  name1: value1
+  name2: value2
+otherInfo: Any other info
+status: alpha
+`);
+}
+```
+
+Keen readers may have noticed the `id: 12345` line in the above code snippet.
+The ID specified as part of the metadata is used for the alerts raised by the script.
+This means that different scripts can now raise alerts with different IDs, which was not possible before.
```

Comment:
It was possible, the `Alert.Builder` did allow to set the ID of the scan rule (`setPluginId(int)`, e.g. https://github.com/zaproxy/zap-extensions/blob/839d5c7df7432c1f4748275daacd1d4e8812a51a/addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Private%20IP%20Disclosure.js#L85).

Reply:
I thought I was missing something. Thanks, will update.